include/zephyr/modbus/modbus.h

```diff
@@ -375,6 +375,11 @@ int modbus_iface_get_by_name(const char *iface_name);
 /**
  * @brief ADU raw callback function signature
  *
+ * This is used for separating the transmission channel from query
+ * handling. It is not intended for implementing vendor specific function
+ * codes, which is also subtly implied from the signature: it only receives
+ * the outgoing ADU as a const pointer.
+ *
```

Comment:
Is it ok if I include them in a separate commit in this PR? I added the docstring to describe the difference between the two functionalities mostly to clarify the implementation for myself, but it could be useful to others as well.

Reply:
I'll probably reword / improve the description a little bit also.

Reply:
Removed this.